### PR TITLE
[rearchitecture] Set up batch execution

### PR DESCRIPTION
**Pull request by jonathanmetzman**

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
TBR

File: src/clusterfuzz/_internal/google_cloud_utils/batch.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  job.allocation_policy = allocation_policy
+  job.labels = {'env': 'testing', 'type': 'container'}
+  job.logs_policy = batch.LogsPolicy()
+  job.logs_policy.destination = batch.LogsPolicy.Destination.CLOUD_LOGGING
+
+  create_request = batch.CreateJobRequest()
+  create_request.job = job
+  job_name = get_job_name()
+  create_request.job_id = job_name
+  # The job's parent is the region in which the job will run
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/google_cloud_utils/batch.py:104]: Can we read these from the config instead?

File: src/clusterfuzz/_internal/google_cloud_utils/batch.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  project_id = 'google.com:clusterfuzz'
+  region = 'us-central1'
+  create_request.parent = f'projects/{project_id}/locations/{region}'
+
+  return _batch_client().create_job(create_request)
+
+
+def main():
+  """Main function only used in development."""
+  email = 'untrusted-worker@clusterfuzz.google.com.iam.gserviceaccount.com'
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/google_cloud_utils/batch.py:114]: Same here. Please read this from the config instead.

File: src/clusterfuzz/_internal/google_cloud_utils/batch.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+def create_job(email,
+               image_uri='gcr.io/clusterfuzz-images/oss-fuzz/worker',
+               machine_type='e2-standard-2'):
+  """This is not a job in ClusterFuzz's meaning of the word."""
+  # Define what will be done as part of the job.
+  runnable = batch.Runnable()
+  runnable.container = batch.Runnable.Container()
+  runnable.container.image_uri = image_uri
+  runnable.container.options = (
+      '--memory-swappiness=40 --shm-size=1.9g --rm --net=host -e HOST_UID=1337 '
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/google_cloud_utils/batch.py:64]: Can we put this in a constant?

File: src/clusterfuzz/_internal/google_cloud_utils/batch.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
+# Copyright 2023 Google LLC
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/google_cloud_utils/batch.py:1]: General comment: we should move a lot of these into our local_config yaml files intead.

File: src/python/bot/startup/run_bot.py [hunk 5]
```diff
@@ -26,6 +26,9 @@
 import time
 import traceback
 
+# !!! How do I import this script.
+import run_batch
```
[the remark this answers is not in the document]
Comment (oliverchang) [reply src/python/bot/startup/run_bot.py]: Does something like from bot.startup import run_batch work?

**Pull request merged**